### PR TITLE
Document system actions 1367

### PR DESCRIPTION
**Pull request by bwatson78**
- app/jobs/README.md: creates Readme file for future Software Engineers.

### CODE REVIEW

**Comment by bwatson78** [comment]
No screenshot necessary.

**Comment by devanshu-m** [comment]
This PR has work that's already in main. Can you please rebase?

File: app/jobs/README.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+### [FileSetActor](https://github.com/emory-libraries/dlp-curate/blob/main/app/actors/hyrax/actors/file_set_actor.rb)
+This is an overwrite of Hyrax' [FileSetActor](https://github.com/samvera/hyrax/blob/v3.0.0.pre.rc1/app/actors/hyrax/actors/file_set_actor.rb). This actor attaches `FileSet`s to a `CurateGenericWork`. Changes that were needed are: 
+- `#create_content`, `#update_content`, and `#wrapper!`: a customized `FileSet` model was utilized, needing a different default `relation` argument set, as well as accepting a newly created argument (`preferred`) that denotes whether a file should be the first piece a user sees.
```
Comment (devanshu-m) [inline app/jobs/README.md:36]: `preferred` is used to specify which is the preferred file for a file_set. The preference can be found here:
https://github.com/emory-libraries/dlp-curate/blob/main/app/models/file_set.rb#L76

File: app/jobs/README.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+- `#attach_to_work`: passing this `file_set` into `work.ordered_members` is omitted because the `ordered_members` association is not utilized in this Actor, but rather in its Job relative.
+- `#fileset_name`: this method was added as a convenience when this Actor is used in other Jobs and RSpec. It makes assigning `FileSet`s `label`s and `title`s easier.
+## Jobs
+### [AttachFilesToWorkJob](https://github.com/emory-libraries/dlp-curate/blob/main/app/jobs/attach_files_to_work_job.rb)
+This is an overwrite of Hyrax' [AttachFilesToWorkJob](https://github.com/samvera/hyrax/blob/v3.0.0.pre.rc1/app/jobs/attach_files_to_work_job.rb). The Job invokes the `FileSetActor` class so that the Actor is available out of the constraints of the Middleware Stack. The following changes were necessary for this Actor:
+- `#process_fileset`: `AttachFilesToWorkJob` had to accept multiple files into one `FileSet`, which Hyrax is not ready for out-of-the-box. This method was created so that multiple types of files could get attached, as well as the `preferred` attribute assigned.
+- `#perform`: the metadata and content creation methods have been moved to the `process_fileset` command which is called here.
```
Comment (devanshu-m) [inline app/jobs/README.md:44]: Were you going for `which is called here` followed by link with line number to the method?

File: app/jobs/README.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+- `#perform`: The generation of thumbnails using MiniMagick has proven problematic and the addition of error logging has been implemented because of it.
+## Bulk Import Processing Sequence
+The following is each procedure in order of start to finish that a CSV uploaded to Curate Dashboard's "Import Content From a CSV" goes through.
+1. [`Zizia::StartCsvImportJob`](https://github.com/curationexperts/zizia/blob/v5.3.0/app/jobs/zizia/start_csv_import_job.rb)
+    The CSV uploaded into the importer gets processed by the Zizia gem. This job is initiated and passes the CSV object to the
+2. [`ModularImporter#import`](https://github.com/curationexperts/zizia/blob/v5.3.0/app/importers/modular_importer.rb)
+    That also resides inside the Zizia Gem. Here, the CSV is completely parsed into a single Rails object using the Gem's built-in parsing and importing tools. Each work parsed from the CSV is assigned to a `record` instance associated with the main Rails `importer` object. Each `record` has a deduplication key and relevant files attached to it, everything is saved, and then passed along. `Zizia::Importer#import` receives it next, which simply passes each record through to
```
Comment (devanshu-m) [inline app/jobs/README.md:58]: Is this point complete? Did you intend to add anything after through to?

File: app/jobs/README.md (hunk 2 above)
Comment (bwatson78) [reply app/jobs/README.md:44]: No, it's just confusing. I need to clear this up.

File: app/jobs/README.md (hunk 3 above)
Comment (bwatson78) [reply app/jobs/README.md:58]: It's just the style of writing I was going for. My intention was to flow action/job into action/job (#1's description flows into #2, and so forth. I think I abandon the convention pretty quick. I'll make edits.

File: app/jobs/README.md (hunk 1 above)
Comment (bwatson78) [reply app/jobs/README.md:36]: I'll expand on this.

**Comment by bwatson78** [comment]
@devanshu-m Changes addressed.

**Pull request merged**